### PR TITLE
SUICIDE VESTS

### PR DESCRIPTION
**Pull request by praisenarsie**
:cl: Frozenguy5
add: Added suicide vests
/:cl:

It compiles but when I try crafting it, it doesn't work. :thinking: help me

### CODE REVIEW

**Comment by yoyobatty** [comment]
So basically you made an override of armor/vest but you never made a custom vest. The requirements are vest but it makes vest.

**Comment by praisenarsie** [comment]
I copied spear/explosive lance code

File: hippiestation/code/modules/clothing/suits/armor.dm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		desc = "A makeshift bomb vest with a [G] attached as a payload."
+	update_icon()
+
+/obj/item/clothing/suit/armor/vest/proc/Trigger(var/mob/living/carbon/human/H)
+	if(H.wear_suit == src)
+		if(H.stat == DEAD)
+			H.visible_message("<span class='danger'>A beep emits from [H]'s vest. It's going to explode!")
+			playsound(loc, 'hippiestation/sound/effects/bombvest_beep.ogg', 50, 1)
+			addtimer(CALLBACK(explosive, /obj/item/grenade.proc/prime, 10))
+			qdel(src)
```
Comment (ghost) [inline hippiestation/code/modules/clothing/suits/armor.dm:33]: i suspect this qdel causes the grenade to delete before the timer can finish
try commenting it out and testing

**Comment by ghost** [comment]
Would make for an interesting(?) traitor item.

+1

Greetings, Fluxcapacitor1337
Head Administrator

**Comment by ghost** [comment]
id say traitor item would be lame as it would only be an accessory for glorious death,  but craftable out of hazard vest and something else would be cool and actually enrich the gameplay than just add another item to the traitor uplink

**Comment by JohnGinnane** [comment]
it'd be a terrible traitor item

**Comment by Manly-Dwarf** [comment]
"I copied spear/explosive lance code"
It's okay. We all steal code here.

**Comment by ghost** [comment]
Maybe a sound effect could accompany the explosion perhaps 2-3 seconds before.

Greetings, Fluxcapacitor1337
Head Administrator

**Comment by ghost** [comment]
*saleelu sawarim...*

**Comment by Eagle-Eyess** [comment]
so why use this instead of something like a lance or syndiebomb in hand?

**Comment by ghost** [comment]
anything politically incorrect gets my useless +1

File: hippiestation/code/modules/clothing/suits/armor.dm (hunk 1 above)
Comment (ghost) [reply hippiestation/code/modules/clothing/suits/armor.dm:33]: or use QDEL_IN(src, 12)

**Comment by Carbonhell** [comment]
>something that works like the most op item currently ingame
I only see this used by powergaming cunts who prefer to damage as much shit as possible when dying, and boring antags.
I do not like the idea a lot. -1

**Comment by Carbonhell** [comment]
needs opinions...

**Pull request closed without merging**